### ISSUE TITLE
queryDistributedElements can return children out of order

### ISSUE THREAD

**Issue by atotic** [Polymer/polymer#2192]
If a node contains `<content> <divs>`,  queryDistributedElements returns a list where `<divs>` come first.

Expected behavior: <divs> should be last. 

Example:
```
<dom-module id="tab-bar">
  <template>
    <iron-selector class="layout horizontal wrap" selected="{{selected}}">
      <content></content>
      <div id="more" icon="icons:more-horiz" text="More"></div>
    </iron-selector>
  </template>
</dom-module>

<tab-bar>
<div id="1">1</div>
<div id="2">2</div>
</tab-bar>
```
`tab-bar.queryDistributedElements()` returns `['#more', '#1', '#2']`
It should return  `['#1', '#2', '#more', ]`

Fix seems pretty easy, inline the first _distributedFilter inside queryDistributedElements and do the right thing.

This is messin' up my fancy responsive tab-bar!

**Comment by kevinpschaaf** [comment]
Repro here: http://jsbin.com/botuxa/edit?html,console,output

**Comment by sorvell** [comment]
Was fixed by at least 1.2.4. Closing.
